unikraft/app/application.go

```diff
@@ -915,11 +916,16 @@ func (app application) WithTarget(targ target.Target) (Application, error) {
 // MarshalYAML makes application implement yaml.Marshaller
 func (app application) MarshalYAML() (interface{}, error) {
 	ret := map[string]interface{}{
-		"spec":     schema.SchemaVersionLatest,
 		"name":     app.name,
 		"unikraft": app.unikraft,
 	}
 
+	if app.kraftfile.config["specification"] != nil {
+		ret["specification"] = schema.SchemaVersionLatest
```

Comment:
Shouldn't this be like this?

```suggestion
	if app.kraftfile.config["specification"] != nil {
		ret["specification"] = app.kraftfile.config["specification"]
```

Reply:
Yes, we can do this 👍🏽